cmd/actuary/dockerhost_test.go

```diff
@@ -0,0 +1,352 @@
+	call string
+	obj []byte
+}
+
+type imageList struct{
+	images []types.Image
+}
+
+type typeContainerList struct{
+	typeContainers []types.Container
+}
+
+//For testing functions that require a specific number of images
+func (list imageList) populateImageList(size int) (imageList){
+	list.images = nil
+	var img types.Image
+	for i := 0; i < size; i++{
+		img = types.Image{ID: strconv.Itoa(i)}
+		list.images = append(list.images, img)
+	}
+	return list
+}
+
+//For testing functions that require a specific number of containers
+func (list typeContainerList) populateContainerList(size int) (typeContainerList){
+	list.typeContainers = nil
+	var c types.Container
+	for i := 0; i < size; i++{
+		c = types.Container{ID: strconv.Itoa(i)}
+		list.typeContainers = append(list.typeContainers, c)
+	}
+	return list
+}
+
+var testTarget, err = NewTarget()
+
+func testServer (t *testing.T, pairings ...callPairing) (server *httptest.Server) { 
+	mux := http.NewServeMux()
+	for _, pair := range pairings {
+		mux.HandleFunc(
+			fmt.Sprintf("/v1.26%s", pair.call),
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request){
+				w.Header().Set("Content-Type", "application/json")
+				w.Write(pair.obj)
+		}))
+	}
+	server = httptest.NewServer(mux)
+	//manipulate testTarget client to point to server
+	testTarget.Client, err = client.NewClient(server.URL, api.DefaultVersion, nil, nil)
+	if err != nil {
+		t.Errorf("Could not manipulate test target client.")
+		return 
+	}
+	return 
+}
+
+//1. host configuration
+//Mainly system file calls
+//Method here: redirect what files the functions are run on -- files created in the /testdata folder. 
+//Do this through the use of global variables, make sure to restore original values
+func TestCheckSeparatePartitionSuccess(t *testing.T){
+	temp := fstab
+	fstab = "testdata/fstabPass" //redefine the global variable
+	res := CheckSeparatePartition(testTarget)
+	if res.Status != "PASS" {
+		t.Errorf("Fstab set to contain /var/lib/docker, should have passed" )
+	}
+	//restore
+	fstab = temp
+}
+
+func TestCheckSeparatePartitionFail(t *testing.T){
+	temp := fstab
+	fstab = "testdata/fstabFail" //redefine the global variable
+	res := CheckSeparatePartition(testTarget)
+	if res.Status == "PASS" {
+		t.Errorf("Fstab does not contain /var/lib/docker, should not have passed" )
+	}
+	//restore
+	fstab = temp
+}
+	
+//Checks info.KernelVersion of target. Fake info within testTarget
+func TestCheckKernelVersionSuccess(t *testing.T) {
+	testTarget.Info.KernelVersion = "4.9.27-moby"
+	res := CheckKernelVersion(testTarget)
+	if  res.Status != "PASS" {
+		t.Errorf("Kernel Version is correct, should have passed." )
+	}
+}
+
+func TestCheckKernelVersionFail(t *testing.T) {
+	testTarget.Info.KernelVersion = "1.9.27-moby"
+	res := CheckKernelVersion(testTarget)
+	if  res.Status == "PASS" {
+		t.Errorf("Kernel Version is incorrect, should not have passed." )
+	}
+}
+
+func TestCheckRunningServices(t *testing.T) {
+	//not mac compatible
```

Comment:
"open /proc/net/tcp: no such file or directory"
This line in the function itself causes an error